### DISCUSSION TITLE
Converting byte arrays to hex, e.g. Thumbprint in Get-DBADbCertificate, or SID in Get-DBALogin

### DISCUSSION THREAD

**Discussion by TallGuyPNW** [dataplat/dbatools#9486] [Q&A]
Hi all,
I realize that this is likely more of a PowerShell question than specifically a dbatools question, but how do I convert byte arrays returned by various dbatools commands into hex values?  
For example:

$dataTable = Get-DbaDbCertificate -sqlinstance serverinstancenamehere |`
        select-object   @{Label="DatabaseName";Expression={$_.Database}},
                        @{Label="CertificateName";Expression={$_.Name}},
                        Owner,
                        Thumbprint

I see rows for each certificate, with values like {111, 93, 151, 17…} for the Thumbprint, but I want to convert that to the hex value 0x6f5d9711...., into the Thumbprint property in $dataTable for each row returned by Get-DBAdbCertificate. Is there some sort of conversion function I can use in-line in the select-object?  Does anyone have an example I can use?
Thanks!

**Comment by 0x7FFFFFFFFFFFFFFF** [comment]
@TallGuyPNW You can use something like this:
```
PS C:\temp> Get-DbaDbCertificate -sqlinstance node1 | select @{N="ThumbprintHex";E={"0x" + (($_.Thumbprint | % { $_.ToString("X2") }) -join "")}}

ThumbprintHex
-------------
0xE4CD195F1728190FACBC14470BBDE4EDBF98F470
0x6DCC0F2D61CDBF15B4E8CBC4EC5C8A8E29112609
0xE68059F2012A810FB6BD97E67E9BD88E1B582959
0x33AD551718345AA7B1B7A3CB3BFEDE1283F5A28A
0x8BEA169086F58A0D3C335CD0B3DB0CD4A364991F
0xAEFBFD4E8E299AC91352C3A5FE35C234B3142105
0x8C969532FB350CC7B6D269FFD7E81D0EC537B690
0xE4CD195F1728190FACBC14470BBDE4EDBF98F470
0xE68059F2012A810FB6BD97E67E9BD88E1B582959
PS C:\temp>
```

**Reply by TallGuyPNW** [reply]
Perfect - thank you!
